scripts/writeReleaseMetadata.mjs

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable no-restricted-syntax,  no-await-in-loop */
+import { simpleGit } from 'simple-git';
+import semver from 'semver';
+import fs from 'fs';
+import path from 'path';
+import { readSvgDirectory } from './helpers.mjs';
+
+const DATE_OF_FORK = '2020-06-08T16:39:52+0100';
+
+const git = simpleGit();
+
+const currentDir = process.cwd();
+const ICONS_DIR = path.resolve(currentDir, '../icons');
+const iconJsonFiles = readSvgDirectory(ICONS_DIR, '.json');
+
+const location = path.resolve(currentDir, '.vitepress/data', 'releaseMetaData.json');
+
+if (fs.existsSync(location)) {
+  fs.unlinkSync(location);
+}
+
+const fetchAllReleases = async () => {
+  await git.fetch('https://github.com/lucide-icons/lucide.git', '--tags');
+
+  return Promise.all(
+    (await git.tag(['-l']))
+      .trim()
+      .split(/\n/)
+      .filter((tag) => semver.valid(tag))
+      .sort(semver.compare),
+  );
+};
+
+const tags = await fetchAllReleases();
+
+const comparisonsPromises = tags.map(async (tag, index) => {
+  const previousTag = tags[index - 1];
+
+  if (!previousTag) return undefined;
+
+  const diff = await git.diff(['--name-status', '--oneline', previousTag, tag]);
+  const files = diff.split('\n').map((line) => {
+    const [status, file, renamedFile] = line.split('\t');
```

Comment:
I understand the motivation behind switching to renames from aliases support, however, this is by no means foolproof and will only work as intended if the renames were done in an exact manner.

Reply:
No that's not exactly true. When squashing PRs to main branch, Git recognizes the renames, it is sometimes not a fully matching. But Git has a special status for this, it is: `R{percentage of the match}`. So he knows it renamed but also 50 matching, This way we can check on this see: line 75 from this file.
The line here above can be undefined, if not set renamed file will not be used.

Reply:
Yes, but it's still not foolproof though; I've seen false positive renames in PRs when an icon was added and another renamed and the wrong icon (the new one) was detected as the original since the SVG code was deemed to be closer.

Reply:
E.g. with https://github.com/lucide-icons/lucide/pull/1028
`git diff --name-status --oneline v0.173.0 v0.174.0` will output:
```
R062    icons/sort-asc.svg      icons/arrow-down-up.svg
R051    icons/sort-asc.json     icons/arrow-down-narrow-wide.json
R062    icons/sort-desc.svg     icons/arrow-down-narrow-wide.svg
R051    icons/sort-desc.json    icons/arrow-up-wide-narrow.json
```

Which is a complete and utter mess, and is quite weird, since I did in fact use `git mv` for these. 🤷‍♂️

Reply:
But wait is it a problem? We only need to check when the version is changed right? So if the renames are not right but are visible in the version change then still the version tag is valid for that specific icon right?

Reply:
Ok I think I got it sorted.

I don't use the rename filename value to detect the renames but only create a object map with all the icon names with created dates and changed date. After that we loop over all the icons and also check the aliases. If aliases exist we apply the created version tag of the alias as well. So I use aliases again.

The metadata is still not 100% correct if we use aliases. There where some renames before we started using aliases and some aliases are intentionally not applied because the old name was used again for other icon design. But I think 97% correctness is ok, let's move on.

BTW, the builds are now respectively 3:30 min, so that pretty neat